stack.hpp

```diff
@@ -7,6 +7,11 @@ class Stack {
 public:
     Stack(std::size_t capacity): storage_(new T[capacity]), capacity_(capacity) {}
 
+    ~Stack(){
+        delete[] storage_;
+        storage_ = nullptr;
```

Comment:
This is as needed as to clean a room in a burning house. This is literally the last line the object lives, so IMHO no need to set the pointer to NULL.